test/Microsoft.AspNet.Razor.Runtime.Test/DefaultTagHelperContentTest.cs

```diff
@@ -0,0 +1,388 @@
+
+            // Assert
+            Assert.Equal(expected, tagHelperContent.ToString());
+        }
+
+        [Fact]
+        public void SetContentClearsExistingContent()
+        {
+            // Arrange
+            var tagHelperContent = new DefaultTagHelperContent();
+            var expected = "Hello World!";
+            tagHelperContent.SetContent("Contoso");
+
+            // Act
+            tagHelperContent.SetContent(expected);
+
+            // Assert
+            Assert.Equal(expected, tagHelperContent.ToString());
+        }
+
+        [Theory]
+        [InlineData("HelloWorld!")]
+        [InlineData("  ")]
+        public void SetContent_WithTagHelperContent_WorksAsExpected(string expected)
+        {
+            // Arrange
+            var tagHelperContent = new DefaultTagHelperContent();
+            var copiedTagHelperContent = new DefaultTagHelperContent();
+            tagHelperContent.SetContent(expected);
+
+            // Act
+            copiedTagHelperContent.SetContent(tagHelperContent);
+
+            // Assert
+            Assert.Equal(expected, copiedTagHelperContent.GetContent());
+        }
+
+        // GetContent
+        [Fact]
+        public void CanGetContent()
+        {
+            // Arrange
+            var tagHelperContent = new DefaultTagHelperContent();
+            var expected = "Hello World!";
+            tagHelperContent.SetContent(expected);
+
+            // Act
+            var actual = tagHelperContent.GetContent();
+
+            // Assert
+            Assert.Equal(expected, actual);
+        }
+
+        [Fact]
+        public void CanAppendContent()
+        {
+            // Arrange
+            var tagHelperContent = new DefaultTagHelperContent();
+            var expected = "Hello World!";
+
+            // Act
+            tagHelperContent.Append(expected);
+
+            // Assert
+            Assert.Equal(expected, tagHelperContent.GetContent());
+        }
+
+        [Fact]
+        public void CanAppendWithoutSet()
+        {
+            // Arrange
+            var tagHelperContent = new DefaultTagHelperContent();
+            var expected = "Hello World!";
+
+            // Act
+            tagHelperContent.Append(expected);
+
+            // Assert
+            Assert.Equal(expected, tagHelperContent.GetContent());
+        }
+
+        [Fact]
+        public void Append_WithTagHelperContent_MultipleAppends()
+        {
+            // Arrange
+            var tagHelperContent = new DefaultTagHelperContent();
+            var copiedTagHelperContent = new DefaultTagHelperContent();
+            var text1 = "Hello";
+            var text2 = " World!";
+            var expected = text1 + text2;
+            tagHelperContent.Append(text1);
+            tagHelperContent.Append(text2);
+
+            // Act
+            copiedTagHelperContent.Append(tagHelperContent);
+
+            // Assert
+            Assert.Equal(expected, copiedTagHelperContent.GetContent());
```

Comment:
nit: doesn't confirm individual writes e.g. `copiedTagHelperContent` could have called `tagHelperContent.GetContent()`.  suggest a `TagHelperContent` mock or (better because it'll work in Core) implementation that `throw`s when anything but `GetEnumerator()` is called.